src/cudamatrix/cu-kernels.cu

```diff
@@ -741,6 +741,16 @@ static void _set_bias_params(Real* v, const Real* a, Real param_1, Real param_2,
   }
 }
 
+template<typename Real, typename OtherReal>
+__global__
+static void _copy_from_vec(int n, const Real* x, int incx, OtherReal* y, int incy) {
+  int32_cuda i = blockIdx.x * blockDim.x + threadIdx.x;
+   // if (blockIdx.y > 0) return;
+
+  if (i < n) {
+    y[1+(i-1)*incy] = (OtherReal) x[1+(i-1)*incx];
```

Comment:
this looks like a bug: 1+(i-1)*incy should be i*incy.  We always use zero-based indexing.
[things like 1+(i-1)*incy might appear only where there was some weird conversion to/from one-based indexing.]